test/e2e/gc/gc_test.go

```diff
@@ -52,4 +56,30 @@ func TestRevisionGC(t *testing.T) {
 	if val := revision.Labels[serving.RoutingStateLabelKey]; val != "active" {
 		t.Fatalf(`Got revision label %s=%q, want="active"`, serving.RoutingStateLabelKey, val)
 	}
+
+	t.Log("Updating the Service to use a different image.")
+	names.Image = test.PizzaPlanet2
+	image2 := pkgTest.ImagePath(names.Image)
+	if _, err := v1test.PatchService(t, clients, resources.Service, rtesting.WithServiceImage(image2)); err != nil {
+		t.Fatalf("Patch update for Service %s with new image %s failed: %v", names.Service, image2, err)
+	}
+
+	t.Log("Service should reflect new revision created and ready in status.")
+	names.Revision, err = v1test.WaitForServiceLatestRevision(clients, names)
+	if err != nil {
+		t.Fatal("New image not reflected in Service:", err)
+	}
+	t.Log("Waiting for Service to transition to Ready.")
+	if err := v1test.WaitForServiceState(clients.ServingClient, names.Service, v1test.IsServiceReady, "ServiceIsReady"); err != nil {
+		t.Fatal("Error waiting for the service to become ready for the latest revision:", err)
+	}
+
+	time.Sleep(5 * time.Second)
```

Comment:
could we poll here rather than sleeping 5 seconds?

Reply:
Maybe we could poll on retrieving the revision (until it comes back as not-exists or times-out). I like that